_posts/2022-XX-XX-configuring-git-repositories.adoc

```diff
@@ -0,0 +1,269 @@
+
+Click *Register application* to create the OAuth App.
+
+Click *Generate new client secret* and save the values of the *Client ID* and *Client Secret* because you will provide them to your {prod-short} administrator.
+
+===== Creating the GitLab Authorized Application
+
+Creating the reference on GitLab is similar to how it's done on GitHub. Additionally, you must choose the allowed scopes.
+
+* Enter *{prod-short}* as the *Name*.
+
+* Enter the {prod-short} OAuth callback URL as the *Redirect URI*. The default value is \https://{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback.
+
+* Check the *Confidential* and *Expire* access tokens checkboxes.
+
+* Under *Scopes*, check the `api`, `write_repository`, and `openid` checkboxes.
+
+Click *Save application* to create the Authorized Application.
+
+Copy and save the *Application ID* and *Secret* values because you will provide them to your {prod-short} administrator.
+
+===== Creating the Bitbucket Application Link
+
+
+Create an RSA key-value pair:
+Use a command line to generate the *public-stripped.pub* and *privatepkcs8-stripped.pem* keys in the required format:
+----
+$ openssl genrsa -out private.pem 2048
+
+$ openssl rsa -in private.pem -pubout > public.pub
+
+$ openssl pkcs8 -topk8 -inform pem -outform pem -nocrypt -in private.pem -out privatepkcs8.pem
+
+$ cat public.pub | sed 's/-----BEGIN PUBLIC KEY-----//g' | sed 's/-----END PUBLIC KEY-----//g' | tr -d '\n' > public-stripped.pub
+
+$ cat privatepkcs8.pem | sed 's/-----BEGIN PRIVATE KEY-----//g' |  sed 's/-----END PRIVATE KEY-----//g' | tr -d '\n' > privatepkcs8-stripped.pem
+----
+
+
+Use a command line to generate a consumer key and a shared secret:
+
+----
+$ openssl rand -base64 24 > bitbucket-consumer-key
+
+$ openssl rand -base64 24 > bitbucket-shared-secret
+----
+
+Create the *Application* link:
+
+Enter the {prod-short} URL as the *Application URL*. The default value is `https://{prod-deployment}-{prod-namespace}.__<domain>__/ ` .
+
+Click *Create new link*.
+
+Under *The supplied Application URL has redirected once*, check the *Use this URL* checkbox and click *Continue*.
+
+Enter the following details for the *Application* link:
+
+* Enter *{prod-short}* as the *Application Name*.
+
+* Select *Generic Application* as the *Application Type*.
+
+* Enter *{prod-short}* as the *Service Provider Name*.
+
+* Paste the content of the *bitbucket-consumer-key* file as the *Consumer key*.
+
+* Paste the content of the *bitbucket-shared-secret* file as the *Shared secret*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/request-token` as the *Request Token URL*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/access-token` as the *Access token URL*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/access-token` as the *Authorize URL*.
```

Comment:
```suggestion
* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/authorize` as the *Authorize URL*.
```

I think it should be `/plugins/servlet/oauth/authorize` like it is in the testing scripts? https://github.com/skabashnyuk/gitsrv/blob/248983e61ae20b47c11396b19ef923136f83ad1c/bitbucket/3_configure_links.sh#L27

Reply:
You are right, good eye. Looks like we had some mistakes in our docs https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-authorization/#proc_configuring-bitbucket-server-oauth1_che